test-integration/helpers/test-element.coffee

```diff
@@ -48,26 +51,37 @@ class TestItemHelper
 
   isPresent: (args...) =>
     locator = @getLocator(args...)
-    @test.driver.isElementPresent(locator).then (isPresent) ->
-      isPresent
+    @test.driver.isElementPresent(locator)
+
+  isDisplayed: (args...) =>
+    @isPresent(args...).then (isPresent) =>
+      if isPresent
+        locator = @getLocator(args...)
+        el = @findElement(args...)
+        el.isDisplayed()
+      else
+        false
 
   # Helper for the common case of `get(...).click()`.
   # Plus, it allows a place to add logging since this is one of the most
   # common places for Selenium to time out (trying to click on an element)
   click: (args...) =>
     locator = @getLocator(args...)
+    # Scroll to the element so it is visible before clicking (this assumes `position: fixed` is overridden for all element)
+    # @isDisplayed(args...).then (isDisplayed) =>
+    #   unless isDisplayed
+    #     el = @findElement(args...)
+    #     @test.utils.windowPosition.scrollTo(el)
     @test.utils.verboseWrap "Clicking #{JSON.stringify(locator)}", =>
-      @get(args...).click()
+      el = @get(args...)
+      @test.utils.windowPosition.scrollTo(el)
+      el.click()
 
-# Using defined properties for access eliminates the possibility
-# of accidental assignment
-Object.defineProperties TestItemHelper.prototype,
-  test:
-    get: -> @_test
-  locator:
-    get: -> @_locator
-  element:
-    get: -> @get()
+  waitClick: (args...) =>
+    locator = @getLocator(args...)
+    @test.utils.verbose "Waiting for #{JSON.stringify(locator)}"
+    @test.utils.wait.for(locator)
+    @click(args...)
 
```

Comment:
@philschatz since `@get` calls `@test.utils.wait.for` underneath, is `waitClick` then trying to wait for the element twice?  Are `click` and `waitClick` actually different?  Could we just add the logging to [`wait`](https://github.com/openstax/tutor-js/blob/master/test-integration/helpers/utils/wait.coffee#L33-L38)'s `click` and make one alias here like this?

```coffee
  waitClick: (args...) =>
    locator = @getLocator(args...)
    @test.utils.wait.click(locator)
```